### PR TITLE
Improve HTTP tests

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Pull Request Description

* simple-httpbin encodes response using the Content-encoding header value
* Return sent body verbatim

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: test/Tests/src/Network/Http_Spec.enso [hunk 1]
```diff
@@ -293,28 +293,14 @@ spec =
...
-                  "url": "",
-                  "method": "POST",
-                  "form": null,
-                  "files": null,
-                  "data": "\uFFFDPNG",
-                  "json": null,
-                  "args": {}
-                }
-            response . should_equal expected_response
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
```
Comment (radeusgd) [inline test/Tests/src/Network/Http_Spec.enso]: Why is the content type `text/plain` here?

Comment (GregoryTravis) [reply test/Tests/src/Network/Http_Spec.enso]: We don't detect the content type of files, but we should.

Comment (GregoryTravis) [reply test/Tests/src/Network/Http_Spec.enso]: Defaulting to application/octet-stream would be good here though.

Comment (radeusgd) [reply test/Tests/src/Network/Http_Spec.enso]: > Defaulting to application/octet-stream would be good here though.

Yep, as long as we cannot detect a more precise content type, I think octet stream is a better default. PNG has nothing to do with txt

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso [hunk 2]
```diff
@@ -213,6 +206,35 @@ fetch (uri:(URI | Text)) (method:HTTP_Method=HTTP_Method.Get) (headers:(Vector (
...
+     - Request_Body.Empty: No content type is sent
+     - Text: `text/plain`
+     - File: `application/octet-stream`
+     - Any other Enso object: `application/json`
+
+   ! Specifying Content Encodings
+
+     The content encoding specified in the `Request_Body` is used to encode the
+     body. The `headers` list can contain additional `Content-Encoding` headers,
+     and these are sent to the server, but not used for encoding.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso]: 👍

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso]: Isn't Content-Encoding for compression? Are we supporting gzip deflate etc?
Its passed as a `charset=...` on the content type.

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

**Review by jdunkerley — CHANGES_REQUESTED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Request_Body.enso [hunk 3]
```diff
@@ -40,6 +41,18 @@ type Request_Body
     ## Empty request body; used for GET
     Empty
 
+    ## PRIVATE
+       For `Request_Body.Form_Data url_encoded=False`, this returns `Nothing`,
+       because the content type must include a content boundary.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Request_Body.enso:46]: I appreciate this comment, I was about to ask why it's Nothing in that case as that was unclear.

**Review by radeusgd — APPROVED** [review]

File: test/Tests/src/Network/Http_Spec.enso (hunk 1 above)
Comment (GregoryTravis) [reply test/Tests/src/Network/Http_Spec.enso]: Done.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso]: Done

File: tools/simple-httpbin/src/main/java/org/enso/shttp/DummyHandler.java [hunk 4]
```diff
@@ -2,22 +2,29 @@
...
 import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.apache.commons.text.StringEscapeUtils;
 
 public class DummyHandler implements HttpHandler {
```
Comment (JaroslavTulach) [inline tools/simple-httpbin/src/main/java/org/enso/shttp/DummyHandler.java]: Why is this class called `DummyHandler`?

File: test/Tests/src/Network/Http_Spec.enso [hunk 5]
```diff
@@ -8,6 +8,7 @@ import Standard.Base.Network.HTTP.Request_Body.Request_Body
 import Standard.Base.Network.HTTP.Request_Error
 import Standard.Base.Network.Proxy.Proxy
 import Standard.Base.Runtime.Context
+from Standard.Base.Network.HTTP import resolve_headers
```
Comment (JaroslavTulach) [inline test/Tests/src/Network/Http_Spec.enso:11]: This test is only executed if:
```
   ## To run this test locally:
       $ sbt 'simple-httpbin/run localhost 8080'
       $ export ENSO_HTTP_TEST_HTTPBIN_URL=http://localhost:8080/
```

Why don't we move the code from `simple-httpbin` into [enso-test-java-helpers](https://github.com/enso-org/enso/tree/d139449bb692a3d122325df6a9220ff021bd71dc/test/Tests/polyglot-sources/enso-test-java-helpers) and let this `Http_Spec` start the server on its own?

**Review by JaroslavTulach — APPROVED** [review]
The title says _"Improve HTTP tests"_ - I didn't know what to expected, but to my disappointment I don't see any simplification of the testing infrastructure. Setting it up still remains complicated and as a result (almost) nobody executes these tests locally.

File: test/Tests/src/Network/Http_Spec.enso (hunk 5 above)
Comment (radeusgd) [reply test/Tests/src/Network/Http_Spec.enso:11]: That sounds like a really cool idea indeed!

Comment (GregoryTravis) [reply test/Tests/src/Network/Http_Spec.enso:11]: I'll break this out as a separate task.

Comment (GregoryTravis) [reply test/Tests/src/Network/Http_Spec.enso:11]: https://github.com/enso-org/enso/issues/7903

**Comment by GregoryTravis** [comment]
@JaroslavTulach The purpose of this was not to improve the test infrastructure, but to provide better testing of `Data.fetch` and `Data.drop`.

**Comment by GregoryTravis** [comment]
@JaroslavTulach Renamed `DummyHandler` to `TestHandler`.

File: tools/simple-httpbin/src/main/java/org/enso/shttp/TestHandler.java [hunk 6]
```diff
@@ -74,6 +82,35 @@ public void handle(HttpExchange exchange) throws IOException {
...
+    } catch (IOException ie) {
+    }
+    try {
+      return baos.toString(encoding);
+    } catch (UnsupportedEncodingException uee) {
+      return "unsupported encoding";
+    }
+  }
+
+  private String parsseTextEncoding(String contentType) {
```
Comment (jdunkerley) [inline tools/simple-httpbin/src/main/java/org/enso/shttp/TestHandler.java]:
```suggestion
  private String parseTextEncoding(String contentType) {
```

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**